### PR TITLE
Added azure script to build dist and wheels and upload on test pass.

### PR DESCRIPTION
**Pull request by nabobalis**
So what this messy of a code block is:

1. Tag release and if its not dev,pre or post in the tag it will trigger an azure build that will run:
  a. the entire suite on Azure (no docs or figure tests)
  b. if they pass will build the wheels and dist and test each one.
  c. if they ALL pass, it will upload them to pypi. 

Things I have not tested:

The actual uploading to pypi.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -1,16 +1,30 @@
...
+  CIBW_TEST_REQUIRES: sqlalchemy scikit-image glymur beautifulsoup4 drms python-dateutil zeep tqdm asdf hypothesis pytest-astropy pytest-mock
 
-jobs:
-- template: azure-templates.yml
-  parameters:
-    name: Linux_36_offline
-    os: linux
-    tox: py36-offline -- -n=4
+trigger:
+  batch: true
```
Comment (Cadair) [inline azure-pipelines.yml]: What does this do?

File: publish-pypi.yml [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        cibuildwheel --output-dir wheelhouse .
+    - task: PublishPipelineArtifact@0
+      inputs: {targetPath: 'wheelhouse', artifactName: 'windows_wheels' }
+
+- job: sdist
+  dependsOn: ${{ parameters.dependsOn }}
+  condition: succeeded()
+  pool: {vmImage: 'Ubuntu-16.04'}
+  steps:
+  - script: |
```
Comment (Cadair) [inline publish-pypi.yml]: arrgh haven't they fixed this yet :roll_eyes:

File: publish-pypi.yml [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  # Remove the ubuntu-toolchain-r/test PPA, which is added by default. Some
+  # packages were removed, and this is causing the g++multilib install to fail.
+  # Similar issue: https://github.com/scikit-learn/scikit-learn/issues/13928
+  - bash: 'sudo add-apt-repository --remove ppa:ubuntu-toolchain-r/test'
+    displayName: "Remove ppa:ubuntu-toolchain-r/test"
+  - script: 'sudo apt-get install -y --fix-missing libblas-dev liblapack-dev'
+    displayName: "Install Scipy deps"
+  - script: 'python -m pip install -U --user --force-reinstall pep517 setuptools_scm'
+    displayName: "acquire build tools"
+  - script: 'python -m pep517.build --source --out-dir wheelhouse .'
```
Comment (Cadair) [inline publish-pypi.yml]: Is it worth explicitly updating the submodule version before here? Or are we always sure we are getting a clean checkout?

File: tox.ini [hunk 4]
```diff
@@ -58,35 +58,3 @@ conda_deps =
     pytest-cov
 conda_channels = conda-forge
 commands = {env:PYTEST_COMMAND} -m "figure" {posargs}
-
-# This env requires tox-conda.
-[testenv:conda]
```
Comment (Cadair) [inline tox.ini]: no conda?

File: azure-pipelines.yml (hunk 1 above)
Comment (nabobalis) [reply azure-pipelines.yml]: `If you set batch to true, when a build is running, the system waits until the build is completed`

Comment (nabobalis) [reply azure-pipelines.yml]: Could remove it.

File: publish-pypi.yml (hunk 2 above)
Comment (nabobalis) [reply publish-pypi.yml]: Nope.

File: publish-pypi.yml (hunk 3 above)
Comment (nabobalis) [reply publish-pypi.yml]: It should be a clean machine that clones and checkouts the submodule.

File: tox.ini (hunk 4 above)
Comment (nabobalis) [reply tox.ini]: Did we ever want it?

Comment (nabobalis) [reply tox.ini]: I might have been over zelous on it

File: publish-pypi.yml (hunk 3 above)
Comment (nabobalis) [reply publish-pypi.yml]: Would you prefer we delete the submodule or manually init it?

Comment (Cadair) [reply publish-pypi.yml]: it should be fine in this context.

File: azure-pipelines.yml [hunk 5]
```diff
@@ -41,9 +54,7 @@ jobs:
...
 
-- ${{ if eq(variables['Build.Reason'], 'Schedule') }}:
-  - template: azure-templates.yml
+- ${{ if startsWith(variables['Build.SourceBranch'], 'refs/tags/') }}:
+  - template: publish-pypi.yml
     parameters:
-      name: Linux_37_numpydev
-      os: linux
-      tox: py37-numpydev -- -n=4
+      dependsOn: [macOS_37_offline,Windows_36_offline,Linux_37_online,Linux_36_Conda_offline,Linux_37_astropydev]
```
Comment (nabobalis) [inline azure-pipelines.yml]: So do we want to depend on the entire azure CI or just the main 3?

File: publish-pypi.yml [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+  - task: UsePythonVersion@0
+    displayName: setup python3.7
+    inputs: {versionSpec: '3.7'}
+  - script: 'python -m pip install -U --user --force-reinstall twine'
+    displayName: "install twine"
+  - task: DownloadPipelineArtifact@1
+  - task: TwineAuthenticate@0
+    inputs:
+      externalFeeds: 'pypi_sunpy'
+  - script: 'python3.7 -m twine upload -r test --config-file $(PYPIRC_PATH) $(System.ArtifactsDirectory)/*/*'
```
Comment (nabobalis) [inline publish-pypi.yml]: This will need to be setup on Azure Web.

**Comment by nabobalis** [comment]
The upload works on the test PyPi: https://test.pypi.org/project/sunpy/1.0.3/ 

Once it was setup on the Azure Web.

File: azure-pipelines.yml (hunk 5 above)
Comment (Cadair) [reply azure-pipelines.yml]: I would drop astropy dev but probably leave the others. All three OSes, online and offline and conda are all important.

File: setup.cfg [hunk 7]
```diff
@@ -80,7 +91,7 @@ testpaths = "sunpy" "docs"
 norecursedirs = ".tox" "build" "docs[\/]_build" "docs[\/]generated" "*.egg-info" "astropy_helpers" "examples" "sunpy[\/]data"
 doctest_plus = enabled
 doctest_optionflags = NORMALIZE_WHITESPACE FLOAT_CMP ELLIPSIS
-addopts = --doctest-rst -m "not figure"
+addopts = -p no:warnings --doctest-rst -m "not figure"
```
Comment (Cadair) [inline setup.cfg]: you adding this back on purpose?

Comment (nabobalis) [reply setup.cfg]: Yes. There are warnings not caused by sunpy in the log and scrolling pasting them (and the sunpy ones) on the CI over and over again was driving me mad.

**Comment by Cadair** [comment]
@nabobalis I will leave the merge button to you on this one.

**Pull request merged**